src/chk/chk_internal.h

```diff
@@ -412,6 +412,24 @@ int chk_iv_init(void);
 
 int chk_iv_fini(void);
 
+/* chk_leader.c */
+
+int chk_leader_report(uint64_t gen, uint32_t cla, uint32_t act, int32_t result, d_rank_t rank,
+		      uint32_t target, uuid_t *pool, uuid_t *cont, daos_unit_oid_t *obj,
+		      daos_key_t *dkey, daos_key_t *akey, char *msg, uint32_t option_nr,
+		      uint32_t *options, uint32_t detail_nr, d_sg_list_t *details, bool local,
+		      uint64_t *seq);
```

Comment:
[Nit] Personally I'd define a struct comprising most of the parameters here, so that it'd be easier to see which argument is for which parameter.

Reply:
OK, I will define a structure to contains these parameters.